### DISCUSSION TITLE
Unite all Destinations into one `NavigationDestination` class

### DISCUSSION THREAD

**Discussion by bleudev** [flet-dev/flet#3104] [Ideas]
**Please Describe The Problem To Be Solved**

Now flet has 3 destinations: `NavigationDestination`, `NavigationDrawerDestination`, `NavigationRailDestination`. I suggest to unite all this destinations because they have almost same arguments (`NavigationRailDestination` has `label_content`, `NavigationDestination` has `tooltip`). But after realisation of this feature for me and other programmers will be very comfortable to use destinations. Also actually you need to improve documentation after realisation

**(Optional): Suggest A Solution**

When user use `NavigationDrawerDestination` f. e. you can just will ignore `label_content` and `tooltip`. Also you can add support for `label_content` in `NavigationBar` and `NavigationDrawer`. Also for `tooltip`.

**Comment by ndonkoHenri** [comment]
Thanks for your suggestion, but I personally don't find this a good idea. :(

Do you know of the [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle) programming principle? 

Combining all of them into one could only be considered if they all have exactly thesame props. As you mentioned, there are differences (presently, and perhaps more in the future). 
When using a class/control, one should normally expect all of the exposed properties to have an effect - this will not be the case.

What I am even thinking of doing now that you raised this issue is to rename `NavigationDestination` to `NavigationBarDestination` for more self-definition and clarity :)

**Reply by bleudev** [reply]
> What I am even thinking of doing now that you raised this issue is to rename `NavigationDestination` to `NavigationBarDestination` for more self-definition and clarity :)

I certainly didn’t open an issue for this, but it’s quite possible to rename NavigationDestination to NavigationBarDestination

**Reply by bleudev** [reply]
I opened pull for that (#3172)
